### ISSUE TITLE
Affero GPLv3 prevents use in many cases

### ISSUE THREAD

**Issue by bhollis** [hannesg/uri_template#12]
I just saw that this library is using the Affero GPLv3 license. That license makes this library unsuitable for use except in very narrow situations (when combined into a compatibly-licensed application that also includes functionality to serve the source code of this library to users).

Ruby code is generally licensed with a more permissive license like MIT, which would not force GPLv3 terms on users of that license.

**Comment by hannesg** [comment]
Hi bhollis

I actually prefer free software over open software and therefore chose GPL deliberately. However, I don't want to keep people from using the software if they have good intents. Do you have a specific usecase?

**Comment by bhollis** [comment]
My intention was to use it as part of the Middleman blog plugin: https://github.com/middleman/middleman-blog, but I don't want to apply the terms of the AGPLv3 to my MIT-licensed software (or to Middleman itself, which is also MIT). Middleman is used in commercial environments where GPLv3 is unacceptable.

**Comment by bhollis** [comment]
Note that this is no longer a huge deal as I have already switched my code to use `addressable`, but I wanted to file this issue just to make you aware that AGPLv3 licensing is in conflict with most Ruby library/application licenses.

**Comment by jasonporritt** [comment]
I recently ran into the AGPLv3 licensing of this gem. I would love to see the license changed to something better-aligned with the Rubygems community at large.

We're using Hyperclient in some batch-processing tasks that interact with an API we've built. Hyperclient is MIT-licensed software, but it carries a dependency on uri_template. While it seems like we _should_ be able to use uri_template after reading through the AGPLv3 text and various interpretations (we're not creating a derivative work, only using the library via its defined API), the license doesn't lend the kind of confidence I'm accustomed to from the Ruby license (2-clause BSD) or MIT license. 

We'd love to continue using it, but may not be able to on account of others' feelings about the license.
